node/src/bootstrap/bootstrap_attempt.rs

```diff
@@ -121,34 +119,26 @@ impl BootstrapAttempt {
             "Starting bootstrap attempt with ID: {id} (mode: {}) ",
             self.mode.as_str()
         );
-        if let Some(websocket) = &self.websocket_server {
-            websocket.broadcast(&self.bootstrap_started());
-        }
-        Ok(())
-    }
 
-    fn bootstrap_started(&self) -> OutgoingMessageEnvelope {
-        OutgoingMessageEnvelope::new(
-            Topic::Bootstrap,
-            BootstrapStarted {
-                reason: "started".to_owned(),
-                id: self.id.clone(),
-                mode: self.mode.as_str().to_owned(),
-            },
-        )
-    }
+        let callbacks = {
+            let callbacks_guard = self
+                .bootstrap_callbacks
+                .bootstrap_started_observer
+                .lock()
+                .unwrap();
+            callbacks_guard.clone()
+        };
+
+        for callback in callbacks.iter() {
+            callback(BootstrapCallbackData::new(
+                self.id.clone(),
+                self.mode,
+                self.total_blocks.load(Ordering::SeqCst),
+                self.duration(),
+            ));
+        }
 
-    fn bootstrap_exited(&self) -> OutgoingMessageEnvelope {
-        OutgoingMessageEnvelope::new(
-            Topic::Bootstrap,
-            BootstrapExited {
-                reason: "exited".to_owned(),
-                id: self.id.clone(),
-                mode: self.mode.as_str().to_owned(),
-                total_blocks: self.total_blocks.load(Ordering::SeqCst).to_string(),
-                duration: self.duration().as_secs().to_string(),
-            },
-        )
+        Ok(())
```

Comment:
Sorry for being so nitpicky, but this behavior belongs into the new BootstrapCallbacks struct. 
A function that we can call... something like:
`self.callbacks.bootstrap_started(BootstrapCallbackData{...});`
The reasons are: better encapsulation, anemic model antipattern, feature envy anti pattern